### DISCUSSION TITLE
Handle error in case the translation file is not available

### DISCUSSION THREAD

**Discussion by lm-rupesh** [ocombe/ng2-translate#1489] [Q&A]
While setting the the current lang using the TranslateService.use() method based on user preference. I want to make sure that if the use lang is not available with use, it should fallback to the default language which will definitely be available.
i.e the new call for say en-GB throws 404, I handle that and again set the currentLang to en-US. 

If we are setting it in app initializer it blocks the loading altogether.

**Comment by CodeAndWeb** [comment]
Please provide a minimal testing app. It's really hard to know what you mean by "setting it in app initializer"...

**Reply by lm-rupesh** [reply]
@ngModule(providers: [{
   provide: APP_INITIALIZER,
      useFactory: initialiseI18nFactory,
      deps: [TranslateService],
      multi: true,}]
      
      function initialiseI18nFactory(translate){
         translate.use(navigator.language).pipe(catchError(err=> translate.use('en-US')))
      }

**Reply by CodeAndWeb** [reply]
What about directly embedding en-US in your app? When setting the default-language, it should fall back to that one.

See https://www.codeandweb.com/babeledit/tutorials/how-to-translate-your-angular-app-with-ngx-translate#ngx-translate-glitches

**Reply by lm-rupesh** [reply]
We have already set the default language in the forRoot configuration and it fetched the default json correctly. Should we move the setting of current language out of the initilizer code?

The other option being I maintain list of available languages in a constant, and checking if the lang is available in the list.
